core/src/main/java/org/apache/brooklyn/core/config/internal/AbstractConfigMapImpl.java

```diff
@@ -107,4 +252,227 @@ public int size() {
+        }
+        return result;
+    }
+    
+    @SuppressWarnings("unchecked")
+    public ReferenceWithError<ConfigValueAtContainer<TContainer,?>> getConfigInheritedRaw(ConfigKey<?> key) {
+        return (ReferenceWithError<ConfigValueAtContainer<TContainer,?>>) (ReferenceWithError<?>) getConfigImpl(key, true);
+    }
+    
+    @Override
+    public Map<ConfigKey<?>, Object> getAllConfigInheritedRawValuesIgnoringErrors() {
+        Map<ConfigKey<?>, ReferenceWithError<ConfigValueAtContainer<TContainer, ?>>> input = getAllConfigInheritedRawWithErrors();
+        Map<ConfigKey<?>, Object> result = MutableMap.of();
+        for (Map.Entry<ConfigKey<?>, ReferenceWithError<ConfigValueAtContainer<TContainer, ?>>> pair: input.entrySet()) {
+            result.put(pair.getKey(), pair.getValue().getWithoutError().get());
+        }
+        return result;
+    }
+    @Override
+    public Map<ConfigKey<?>, ReferenceWithError<ConfigValueAtContainer<TContainer, ?>>> getAllConfigInheritedRawWithErrors() {
+        return getSelectedConfigInheritedRaw(false);
+    }
+    
+    public Map<ConfigKey<?>,ReferenceWithError<ConfigValueAtContainer<TContainer,?>>> getAllReinheritableConfigRaw() {
+        return getSelectedConfigInheritedRaw(true);
+    }
+
+    protected Map<ConfigKey<?>,ReferenceWithError<ConfigValueAtContainer<TContainer,?>>> getSelectedConfigInheritedRaw(boolean onlyReinheritable) {
+        Map<ConfigKey<?>,ReferenceWithError<ConfigValueAtContainer<TContainer,?>>> parents = MutableMap.of();
+        if (getParent()!=null) {
+            @SuppressWarnings("unchecked")
+            Map<ConfigKey<?>,ReferenceWithError<ConfigValueAtContainer<TContainer,?>>> po = (Map<ConfigKey<?>,ReferenceWithError<ConfigValueAtContainer<TContainer,?>>>) (Map<?,?>)
+                getParentInternal().config().getInternalConfigMap().getAllReinheritableConfigRaw();
+            parents.putAll(po);
+        }
+        
+        Map<ConfigKey<?>, Object> local = getAllConfigLocalRaw();
+        
+        Map<ConfigKey<?>,ReferenceWithError<ConfigValueAtContainer<TContainer,?>>> result = MutableMap.of();
+
+        for (ConfigKey<?> k: local.keySet()) {
+            Object v = local.get(k);
+            ReferenceWithError<ConfigValueAtContainer<TContainer, ?>> vpr = parents.remove(k);
+            @SuppressWarnings("unchecked")
+            ConfigValueAtContainer<TContainer, Object> vp = vpr==null ? null : (ConfigValueAtContainer<TContainer,Object>) vpr.getWithoutError();
+            ConfigInheritance inh = ConfigInheritances.findInheritance(k, InheritanceContext.RUNTIME_MANAGEMENT, getDefaultRuntimeInheritance());
+            ConfigValueAtContainer<TContainer,Object> vl = new BasicConfigValueAtContainer<TContainer,Object>(getContainer(), k, Maybe.ofAllowingNull(v));
+            ReferenceWithError<ConfigValueAtContainer<TContainer, Object>> vlr = null;
+            if (inh.considerParent(vl, vp, InheritanceContext.RUNTIME_MANAGEMENT)) {
```

Comment:
as per javadoc methods in `ConfigInheritances` do that; the methods here are to reflect the component logic which needs to be defined for a consistent inheritance strategy